### PR TITLE
Add limit to participants

### PR DESCRIPTION
**Pull request by ZePedroResende**

### CODE REVIEW

File: app/models/activity.rb [hunk 1]
```diff
@@ -39,6 +40,10 @@ def registered?(user)
     registrations.find_by(user_id: user.id)
   end
 
+  def can_register?
+    participants.count < number_participants
+  end
```
Comment (9thScientist) [inline app/models/activity.rb]: Shouldn't this be a helper method?

File: app/views/activities/_form.slim [hunk 2]
```diff
@@ -30,10 +30,11 @@
   = form.text_field :speaker,
     class: "activity-form-field"
 
-  = form.label :total_rating,
+  = form.label :number_participants,
     class: "activity-form-label"
   br
-  = form.text_field :total_rating,
+  = form.number_field :number_participants,
+    value: 20,
```
Comment (9thScientist) [inline app/views/activities/_form.slim]: You are setting by default, the maximum number of participants as 20. But what if we don't want to specify the maximum number of participants?

File: config/locales/pt.yml [hunk 3]
```diff
@@ -25,3 +25,12 @@ pt:
...
     next_label: "Seguinte"
+
+  activerecord:
+    errors:
+      models:
+        activity:
+          attributes:
+            number_participants:
+              blank: "vazio"
+              not_a_number: "nao e um numero"
```
Comment (9thScientist) [inline config/locales/pt.yml]: Please, respect the language of Camões.

File: app/views/activities/_form.slim (hunk 2 above)
Comment (ZePedroResende) [reply app/views/activities/_form.slim]: You can give 0 to have no limite

File: config/locales/pt.yml (hunk 3 above)
Comment (ZePedroResende) [reply config/locales/pt.yml]: :'( 9.5 in Portuguese

File: app/models/activity.rb [hunk 4]
```diff
@@ -39,6 +40,10 @@ def registered?(user)
     registrations.find_by(user_id: user.id)
   end
 
+  def can_register?
```
Comment (marcelo140) [inline app/models/activity.rb]: The name seems too vague for me. It should be something more related to the fact that the talk isn't full. Something along the lines `has_sits_available?`

File: app/views/activities/_form.slim (hunk 2 above)
Comment (marcelo140) [reply app/views/activities/_form.slim]: I don't like this neither. If it's unlimited, it should be left blank.

File: app/controllers/activities/registrations_controller.rb [hunk 5]
```diff
@@ -3,8 +3,10 @@ class Activities::RegistrationsController < ApplicationController
   load_and_authorize_resource
 
   def create
-    unless @activity.registered?(current_user)
-      Registration.create(activity_id: @activity.id, user_id: current_user.id)
+    Activity.transaction do
+      if !@activity.registered?(current_user) && @activity.can_register?
+        Registration.create(activity_id: @activity.id, user_id: current_user.id)
+      end
```
Comment (marcelo140) [inline app/controllers/activities/registrations_controller.rb]: This should be a method in the model.

**Comment by ZePedroResende** [comment]
@9thScientist  @marcelo140  ping, i think its all done

**Pull request closed without merging**